clickhouse_connect/driver/httpclient.py

```diff
@@ -283,6 +291,12 @@ def _query_with_context(self, context: QueryContext) -> QueryResult:
         byte_source = RespBuffCls(ResponseSource(response))  # pylint: disable=not-callable
         context.set_response_tz(self._check_tz_change(response.headers.get('X-ClickHouse-Timezone')))
         query_result = self._transform.parse_response(byte_source, context)
+        if self._column_renamer:
```

Comment:
It would be nice to move this functionality into the _transform.parse_response method or the query_result object.   The client doesn't need to know how to modify query_result data, and the _column_renamer seems like it belongs more to the context.  The only annoyance is that the "LIMIT 0" hack is currently in the client because ClickHouse Native won't return column names :(